sakuraio/software/base.py

```diff
@@ -0,0 +1,88 @@
+import requests
+from sakuraio.software.apis import APIMixins
+
+
+# Please check bottom of API docs.
+API_VERSION = "1.0.3"
+
+
+class SakuraIOClient(APIMixins):
+    base_url = 'https://api.sakura.io/v1/'
+    api_token = None
+    api_secret = None
+
+    def __init__(self, api_token, api_secret, base_url=None):
+        if api_token is None or api_secret is None:
+            Exception('[ERROR] must set api_token and api_secret')
+
+        self.api_token = api_token
+        self.api_secret = api_secret
+
+        if base_url is not None:
+            self.base_url = base_url
```

Comment:
When user set `base_url`, he might set wrong `base_url` format, because of typo.
such a `https:/api.sakura.io/v1`...or `https://`
so how about adding some validation for `base_url` format. It is enough that only parse `http` library.

Reply:
`requests` module check url scheme. if input wrong `base_url` , he output Exception (`InvalidSchema`) .

```
(venv) whywaita% cat test.py
from sakuraio.software.base import SakuraIOClient

# wrong scheme
client = SakuraIOClient('***', '***', base_url='htttps://api.sakura.io/v1/')
res = client.show_projects()
print(res)

(venv) whywaita% python test.py
Traceback (most recent call last):
  File "test.py", line 4, in <module>
    res = client.show_projects()
  File "~/git/python-sakuraio/sakuraio/software/apis/project.py", line 12, in show_projects
    query_params={'name': name, 'sort': sort}
  File "~/git/python-sakuraio/sakuraio/software/base.py", line 57, in do
    auth=auth
  File "~/git/python-sakuraio/venv/lib/python2.7/site-packages/requests/api.py", line 72, in get
    return request('get', url, params=params, **kwargs)
  File "~/git/python-sakuraio/venv/lib/python2.7/site-packages/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "~/git/python-sakuraio/venv/lib/python2.7/site-packages/requests/sessions.py", line 502, in request
    resp = self.send(prep, **send_kwargs)
  File "~/git/python-sakuraio/venv/lib/python2.7/site-packages/requests/sessions.py", line 606, in send
    adapter = self.get_adapter(url=request.url)
  File "~/git/python-sakuraio/venv/lib/python2.7/site-packages/requests/sessions.py", line 697, in get_adapter
    raise InvalidSchema("No connection adapters were found for '%s'" % url)
requests.exceptions.InvalidSchema: No connection adapters were found for 'htttps://api.sakura.io/v1/projects/'
```

If you think need more validate, I will add that. Please teach me your idea.

Reply:
Thanks for giving examples.
If `requests` module looks  working enogh.